src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/share/mappers/PathValidationDataMapper.java

```diff
@@ -36,6 +41,10 @@ public abstract class PathValidationDataMapper {
      * @return the messaging representation of a path validation data
      */
     public PathValidationData toPathValidationData(PathValidationPayload pathValidationPayload) {
+        if (pathValidationPayload.getNodes() == null || pathValidationPayload.getNodes().isEmpty()) {
```

Comment:
We could also use here:
`CollectionUtils.isEmpty(pathValidationPayload.getNodes())`
from org.apache.commons.collections4.CollectionUtils